yang/IETF-04/ietf-vpn-common.yang

```diff
@@ -1003,20 +1003,32 @@ module ietf-vpn-common {
   }
 
   grouping rt-rd {
-    leaf rd {
-      type union {
-        type rt-types:route-distinguisher;
-        type empty;
-      }
+      choice rd-choice {
+        case directly-assigned {
+          leaf rd {
+              type rt-types:route-distinguisher;
+            }
+          description
+            "Route distinguisher assigment";
+          }
+          case pool-assigned {
+            leaf rd-pool-name {
+              type string;
+            }
+            description
+              "The server will auto-assign a route
```

Comment:
should meantion that the assignment will be from the pool in the "rd-pool-name".

Reply:
Latest idea, for the pool-assigned option, I was thinking to add the option to let the controller to chose the pool automatically.

      case pool-assigned {
        leaf rd-pool-name {
          type union {
            type string;
            type enumeration {
              enum AUTO;
            }
          }
        }
      }